pydot factory test broken by whitespace changes in dot

Somewhere between graphviz 2.40.1 and 2.42.2, whitespace changes in have occurred in the dot output which break the factory test.

[graphviz 2.42.2](https://formulae.brew.sh/formula/graphviz) is now available in homebrew for both macOS Sierra (Crystal, Dashing), and Mojave (Eloquent). It's also currently in Debian Sid which means that it will become part of 20.04 for future ROS 2 releases.

The actual failure message from [this build](https://ci.ros2.org/job/test_ci_osx/172/testReport/junit/qt_dotgraph.test.pydot_factory_test/PyDotFactoryTest/test_create_dot/):

```
AssertionError: False is not true : foo	 [ 
missing in
 digraph graphname {
	graph [bb=&quot;0,0,54,86&quot;,
		compound=True,
		rank=same,
		rankdir=TB,
		ranksep=0.2
	];
	node [label=&quot;\N&quot;];
	subgraph cluster_foo {
		graph [compound=True,
			label=foo,
			rank=same,
			rankdir=TB,
			ranksep=0.2,
			style=bold
		];
	}
	foo	[height=0.5,
		label=foo,
		pos=&quot;27,68&quot;,
		shape=box,
		width=0.75];
	edge_	[height=0.5,
		label=edge_,
		pos=&quot;27,18&quot;,
		shape=box,
		width=0.75];
	foo -&gt; edge_	[penwidth=1,
		pos=&quot;e,27,36.003 27,49.942 27,48.669 27,47.367 27,46.053&quot;];
}
```

https://github.com/ros-visualization/qt_gui_core/blob/5de77352b48ff5724e521cac19bfdfad0b7a1aa8/qt_dotgraph/test/pydot_factory_test.py#L90-L116

--- a/qt_dotgraph/test/pydot_factory_test.py
+++ b/qt_dotgraph/test/pydot_factory_test.py
@@ -31,6 +31,7 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import re
 import unittest
 from qt_dotgraph.pydotfactory import PydotFactory
 
@@ -93,20 +94,22 @@ class PyDotFactoryTest(unittest.TestCase):
         fac.add_node_to_graph(g, 'edge')
         fac.add_edge_to_graph(g, 'foo', 'edge')
         fac.add_subgraph_to_graph(g, 'foo')
-        snippets = ['digraph graphname {\n\tgraph [',
+        snippets = ['digraph graphname { graph [',
                     'rankdir=TB',
                     'compound=True',
                     'rank=same',
                     'node [label="\\N"]',
-                    'subgraph cluster_foo {\n\t\tgraph [',
-                    'foo\t [',
+                    'subgraph cluster_foo { graph [',
+                    'foo [',
                     'label=foo',
                     'shape=box',
                     'pos="',
-                    'edge_\t [',
+                    'edge_ [',
                     'label=edge_',
-                    'foo -> edge_\t [',
-                    '"];\n}\n']
+                    'foo -> edge_ [',
+                    '"]; }']
         result = fac.create_dot(g)
+        # get rid of version specific whitespaces
+        result = re.sub('[\n\t ]+', ' ', result)
         for sn in snippets:
             self.assertTrue(sn in result, '%s \nmissing in\n %s' % (sn, result))

